packages/mermaid/src/docs/community/code.md

```diff
@@ -0,0 +1,163 @@
+pnpm test
+```
+
+When writing new tests, it's easier to have the tests automatically run as you make changes. You can do this by running the following command:
+
+```sh
+pnpm test:watch
+```
+
+## Integration/End-to-End (e2e) tests
+
+These test the rendering and visual appearance of the diagrams.
+This ensures that the rendering of that feature in the e2e will be reviewed in the release process going forward. Less chance that it breaks!
+
+To start working with the e2e tests:
+
+1. Run `pnpm dev` to start the dev server
+2. Start **Cypress** by running `pnpm cypress:open`.
+
+The rendering tests are very straightforward to create. There is a function `imgSnapshotTest`, which takes a diagram in text form and the mermaid options, and it renders that diagram in Cypress.
+
+When running in CI it will take a snapshot of the rendered diagram and compare it with the snapshot from last build and flag it for review if it differs.
+
+This is what a rendering test looks like:
+
+```js
+it('should render forks and joins', () => {
+  imgSnapshotTest(
+    `
+    stateDiagram
+    state fork_state &lt;&lt;fork&gt;&gt;
+      [*] --> fork_state
+      fork_state --> State2
+      fork_state --> State3
+
+      state join_state &lt;&lt;join&gt;&gt;
+      State2 --> join_state
+      State3 --> join_state
+      join_state --> State4
+      State4 --> [*]
+    `,
+    { logLevel: 0 }
+  );
+  cy.get('svg');
+});
+```
+
+**_[TODO - running the tests against what is expected in development. ]_**
+
+**_[TODO - how to generate new screenshots]_**
+....
+
+## 3. Update Documentation
+
+If the users have no way to know that things have changed, then you haven't really _fixed_ anything for the users; you've just added to making Mermaid feel broken.
+Likewise, if users don't know that there is a new feature that you've implemented, it will forever remain unknown and unused.
+
+The documentation has to be updated to users know that things have changed and added!
+If you are adding a new feature, add `(v<MERMAID_RELEASE_VERSION>+)` in the title or description. It will be replaced automatically with the current version number when the release happens.
+
+eg: `# Feature Name (v<MERMAID_RELEASE_VERSION>+)`
+
+We know it can sometimes be hard to code _and_ write user documentation.
+
+Our documentation is managed in `packages/mermaid/src/docs`. Details on how to edit is in the [Contributing Documentation](#contributing-documentation) section.
+
+Create another issue specifically for the documentation.  
+You will need to help with the PR, but definitely ask for help if you feel stuck.
+When it feels hard to write stuff out, explaining it to someone and having that person ask you clarifying questions can often be 80% of the work!
+
+When in doubt, write up and submit what you can. It can be clarified and refined later. (With documentation, something is better than nothing!)
+
+## 4. Submit your pull request
+
+**[TODO - PR titles should start with (fix | feat | ....)]**
```

Comment:
I think if the branch is called `fix/` or `feat/`, then https://github.com/mermaid-js/mermaid/blob/develop/.github/pr-labeler.yml will automatically label the PR appropriately.

The `fix:`, `feat:` comes from [conventional commits], which are generally used for commit titles. Personally, I'd love it if Mermaid switched to conventional commits, but I think I'm the only @mermaid-js contributor that tries make nice commits, and getting everybody else to switch might be difficult, since then everybody would need to learn how to use `git rebase`.

Reply:
This is simply copy and paste from previous document

Reply:
Okay, it's fine then :) We probably should remove them though in a future PR, since we probably shouldn't have TODO comments that are visible to users in the documentation.

Reply:
Can't we hide ("comment out") the comments ?

Reply:
On the other hand, users knowing this isn't the final docs might also be a good thing. 😅